fml/task_runner_merger.h

```diff
@@ -0,0 +1,47 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FML_SHELL_COMMON_TASK_RUNNER_MERGER_H_
+#define FML_SHELL_COMMON_TASK_RUNNER_MERGER_H_
+
+#include "flutter/fml/macros.h"
+#include "flutter/fml/memory/ref_counted.h"
+#include "flutter/fml/message_loop_task_queues.h"
+
+namespace fml {
+
+class MessageLoopImpl;
+
+class TaskRunnerMerger : public fml::RefCountedThreadSafe<TaskRunnerMerger> {
+ public:
+  void MergeWithLease(size_t lease_term);
+
+  void ExtendLease(size_t lease_term);
+
+  // Returns true if we unmerged this turn.
```

Comment:
nit: "turn" is a little ambiguous, maybe "... if this call resulted in splitting the gpu and platform threads".

Reply:
Done